src/components/displayDocuments/displayDocuments.tsx

```diff
@@ -1,47 +1,85 @@
-import React, { ReactElement, useState, useEffect } from "react";
+import React, { useState, useEffect } from "react";
+import { useParams, Switch, Route, Redirect } from "react-router-dom";
+import { constantCase } from "change-case";
 import { cssContainerWrapper } from "../../constants";
 import { useFuzzy } from "react-use-fuzzy";
 import { MenuBar } from "./menuBar/menuBar";
 import { DocumentCard } from "./documentCard";
-import { Tag, Document } from "./documents";
-interface DisplayDocumentsProps {
+import { Tag, TagType, Document } from "./documents";
+
+interface FilteredDocumentsProps {
+  searchValue: string;
   documents: Document[];
 }
 
-export const DisplayDocuments: React.FunctionComponent<DisplayDocumentsProps> = ({
+const NoDocumentsFound: React.FunctionComponent = () => {
+  return (
+    <div className="text-center w-auto mx-auto py-8">
+      <h2>No documents found.</h2>
+      <p>Try searching another term?</p>
+    </div>
+  );
+};
+
+const FilteredDocuments: React.FunctionComponent<FilteredDocumentsProps> = ({
+  searchValue,
   documents
-}: DisplayDocumentsProps) => {
-  const [selectedButton, setSelectedButton] = useState<Tag | undefined>();
-  const [searchValue, setSearchValue] = useState("");
+}: FilteredDocumentsProps) => {
+  const { tagId } = useParams<{ tagId: string }>();
   const { result, search: fuzzySearch } = useFuzzy<Document>(documents, {
     keys: ["title"]
   });
 
-  useEffect(() => {
-    fuzzySearch(searchValue);
-  }, [fuzzySearch, searchValue]);
-
-  const displayCards = (documents: Document[]): ReactElement[] => {
-    return documents
-      .filter(document => !selectedButton || document.tags.includes(selectedButton))
+  const displayFilteredDocuments = (documents: Document[]): React.ReactNode => {
+    const filtered = documents
+      .filter(document => {
+        if (!tagId) return document;
+        const toMatch = constantCase(tagId);
+        const matched = Object.keys(Tag).find(key => {
+          return key === toMatch;
+        }) as TagType;
+        return document.tags.includes(Tag[matched]);
```

Comment:
doesn´t

```ts
return document.tags.includes(constantCase(tagId));
```
work ?

we transform to the tag, then search the tag, can use the transform value straight no ?